File: app/src/main/java/com/example/revroom/data/remote/ApiService.java
package com.example.revroom.data.remote;

import retrofit2.Call;
import retrofit2.http.DELETE;
import retrofit2.http.GET;
import retrofit2.http.Header;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface ApiService {
    @GET("api/design/projects")
    Call<ProjectHistoryResponse> getProjects(
        @Header("user-id") String userId,
        @Query("page") int page,
        @Query("pageSize") int pageSize
    );

    @DELETE("api/design/{designId}")
    Call<DeleteResponse> deleteProject(
        @Header("user-id") String userId,
        @Path("designId") String designId
    );
}

File: app/src/main/java/com/example/revroom/data/remote/ApiService.kt
package com.example.revroom.data.remote

import okhttp3.MultipartBody
import okhttp3.RequestBody
import retrofit2.Response
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.Multipart
import retrofit2.http.POST
import retrofit2.http.Part
import retrofit2.http.Path

// 1. Tạo các cục hàng để gửi đi / nhận về
data class DeviceRequest(val userId: String)

data class UserProfileResponse(
    val userId: String,
    val name: String?,
    val avatarUrl: String?,
    val createdAt: String?
)

// 2. Khai báo cổng API (CHỈ CÓ 1 INTERFACE NÀY THÔI NHÉ)
interface ApiService {

    // Bắn ID
    @POST("api/Auth/register-device")
    suspend fun registerDevice(@Body request: DeviceRequest): Response<Any>

    // Bắn Ảnh
    @Multipart
    @POST("api/Auth/upload-avatar")
    suspend fun uploadAvatar(
        @Part("userId") userId: RequestBody,
        @Part file: MultipartBody.Part
    ): Response<Any>

    // Kéo thông tin User (kèm link ảnh) về
    @GET("api/Auth/profile/{userId}")
    suspend fun getProfile(@Path("userId") userId: String): Response<UserProfileResponse>
}